sdk/nodejs/resource.ts

```diff
@@ -782,7 +783,22 @@ export class ComponentResource extends Resource {
         // not correspond to a real piece of cloud infrastructure.  As such, changes to it *itself*
         // do not have any effect on the cloud side of things at all.
         super(type, name, /*custom:*/ false, /*props:*/ {}, opts);
-        this.__pulumiComponentResource = true;
+
+        if (this.isAsyncConstructed()) {
+            this.__isConstructed = new Promise(resolve => {
+                this.__resolveIsConstructed = resolve;
+            });
+        }
+    }
+
+    /**
+     * If this is a component that will construct its children  asynchronously. Components that do
+     * this *must* call [registerOutputs] to signal when construction is complete.  If
+     * [registerOutputs] is not called then hangs will occur if another resource 'dependsOn' this
```

Comment:
Is there any way to detect this situation?

Reply:
not really no. how can we detect teh difference between the user's code being particularly slow, versus it forgetting to set this?  we could set a timeout... but it's risky.